src/session_impl.cpp

```diff
@@ -3145,7 +3133,8 @@ namespace aux {
 			for (auto& ext : m_ses_extensions)
```

Comment:
Thank you, I understand and agree with you.

Having said that, I "literally" speculate that maybe in this case, the vector is so small, that it fits in the cache. The iteration "should" still be fast. Btw, I want to comment that with this change, there is a big saving in not calling `on_alerts` and with the `m_session_extension_features` variable, the saving is even bigger if no plugin happens to implement an specific feature (that's yours).

Well, not sure if you are willing to accept this solution, or you have something else in mind.